_pages/dev/extending-checkout.md

```diff
@@ -101,36 +101,113 @@ Every checkout component is a CMS component. Furthermore, in the default checkou
+
+```typescript
+@Injectable({
+  providedIn: "root",
+})
+export class CustomCheckoutStepsSetGuard extends CheckoutStepsSetGuard {
+  constructor(
+    protected checkoutStepService: CheckoutStepService,
+    protected routingConfigService: RoutingConfigService,
+    protected checkoutDeliveryAddressFacade: CheckoutDeliveryAddressFacade,
+    protected checkoutPaymentFacade: CheckoutPaymentFacade,
+    protected checkoutDeliveryModesFacade: CheckoutDeliveryModesFacade,
+    protected router: Router
+  ) {
+    super(
+      checkoutStepService,
+      routingConfigService,
+      checkoutDeliveryAddressFacade,
+      checkoutPaymentFacade,
+      checkoutDeliveryModesFacade,
+      router
+    );
+  }
+
+  protected isDeliveryModeSet(
+    step: CheckoutStep
+  ): Observable<boolean | UrlTree> {
+    // insert custom logic
+    // ...
+  }
+}
+```
+
+Note: make sure to place the dependency provider and have it use the `CustomCheckoutStepsSetGuard` in your module.
+
+```typescript
+@NgModule({
+  ...
+  providers: [
+    {
+      provide: CheckoutStepsSetGuard,
+      useClass: CustomCheckoutStepsSetGuard
+    }
+  ]
+})
+export class SomeModuleName {}
+```
 
 ### CheckoutGuard
 
 A special `CheckoutGuard` is responsible for redirecting to the correct step. The default implementation redirects every checkout request to the first step. You can replace this with your own guard (for example, you could redirect users to the first step that is not set). A custom example is provided in the [Express Checkout](#express-checkout) section, below.
 
+### CheckoutAuthGuard
+
+`CheckoutAuthGuard` is responsible for handling guest and authenticated users to allow them to checkout. However, if they are an anonymous user, it will redirect the user to the login page.
+
+### CartNotEmptyGuard
+
+`CartNotEmptyGuard` is responsible for redirecting the user if they are trying to visit a checkout route when there active cart is currently empty. If the active cart is empty, it redirects the user to homepage, else it will allow them to go through the checkout flow.
+
+### NotCheckoutAuthGuard
+
+- TBD (pretty redundant I think)
+
 ### Configuring Where Guards Redirect To
 
 In the checkout configuration, for each step, you specify a `type` attribute and the type of data that should be set. A guard looks for the first step that contains the specific `type` and then redirects to this step.
 
-For example, `ShippingAddressSetGuard` searches the checkout configuration for the first step with a `type` containing `CheckoutStepType.shippingAddress`, then reads the step route and redirects to that page.
+For example, `CheckoutStepsSetGuard` searches the checkout configuration for every step with a `type` containing `CheckoutStepType.deliveryAddress`, `CheckoutStepType.deliveryMode`, `CheckoutStepType.paymentDetails`, `CheckoutStepType.reviewOrder`, then reads the step route and redirects to that page.
 
```

Comment:
In the updated text, it talks about searching "for every step with a `type` containing...", but in the paragraph above, the text says the guard "looks for the first step that contains the specific `type` and then..."

Is this part still true, and does it need updating to match the text that follows?

Reply:
Before we had like `ShippingAddressSetGuard`, `SomethingSetGuard`, `EtcSetGuard`, but they had been consolidated into one, which is now called `CheckoutStepsSetGuard`. So the old text is not true as it now checks at every step with those types instead of only one.

Reply:
In line 131, I've modified the text so that it says "A guard looks for every step that contains the specific `type` and then redirects to this step."